kate/recovery/src/com.rs

```diff
@@ -706,81 +541,86 @@ mod tests {
+			size: 16,
+			index: vec![(1, 2), (2, 5), (3, 8)],
 		};
-		let dimensions = ExtendedMatrixDimensions { rows: 4, cols: 4 };
-
-		let expected_0 = vec![(0, 0), (0, 2), (1, 0), (1, 2), (2, 0)];
-		let result_0 = app_specific_cells(&index, &dimensions, 0).unwrap();
-
-		assert_eq!(expected_0.len(), result_0.len());
-		result_0.iter().zip(expected_0).for_each(|(a, (col, row))| {
-			assert_eq!(a.col, col);
-			assert_eq!(a.row, row);
-		});
-
-		let expected_1 = vec![(2, 2), (3, 0), (3, 2)];
-		let result_1 = app_specific_cells(&index, &dimensions, 1).unwrap();
-
-		assert_eq!(expected_1.len(), result_1.len());
-		result_1.iter().zip(expected_1).for_each(|(a, (col, row))| {
-			assert_eq!(a.col, col);
-			assert_eq!(a.row, row);
-		});
-
-		assert!(app_specific_cells(&index, &dimensions, 2).is_none());
+		let dimensions = Dimensions::new(8, 4);
+		let result = app_specific_rows(&index, &dimensions, app_id);
+		assert_eq!(expected.len(), result.len());
 	}
 
-	#[test]
-	fn test_app_specific_column_cells() {
+	#[test_case(0, &[(0, 0), (0, 1), (0, 2), (0, 3), (2, 0)] ; "App 0 has five cells")]
+	#[test_case(1, &[(2, 1), (2, 2), (2, 3)] ; "App 1 has 3 cells")]
+	#[test_case(2, &[] ; "App 2 has no cells")]
+	fn test_app_specific_cells(app_id: u32, expected: &[(u32, u16)]) {
 		let index = AppDataIndex {
 			size: 8,
 			index: vec![(1, 5)],
 		};
-		let dimensions = ExtendedMatrixDimensions { rows: 4, cols: 4 };
-
-		let expected_0 = (0..=2).flat_map(|c| (0..=3).map(move |r| (c, r)));
-		let result_0 = app_specific_column_cells(&index, &dimensions, 0).unwrap();
-
-		assert_eq!(expected_0.clone().count(), result_0.len());
-		result_0.iter().zip(expected_0).for_each(|(a, (col, row))| {
-			assert_eq!(a.col, col);
+		let dimensions = Dimensions::new(4, 4);
+		let result = app_specific_cells(&index, &dimensions, app_id).unwrap_or_default();
+		assert_eq!(expected.len(), result.len());
+		result.iter().zip(expected).for_each(|(a, &(row, col))| {
 			assert_eq!(a.row, row);
-		});
-
-		let expected_1 = (2..=3).flat_map(|c| (0..=3).map(move |r| (c, r)));
-		let result_1 = app_specific_column_cells(&index, &dimensions, 1).unwrap();
-
-		assert_eq!(expected_1.clone().count(), result_1.len());
-		result_1.iter().zip(expected_1).for_each(|(a, (col, row))| {
 			assert_eq!(a.col, col);
-			assert_eq!(a.row, row);
 		});
-
-		assert!(app_specific_column_cells(&index, &dimensions, 2).is_none());
 	}
 
-	#[test]
-	fn test_app_specific_column_cells_gt_chunk_size() {
-		let index = AppDataIndex {
-			size: 90,
-			index: vec![(1, 1)],
-		};
-		let dimensions = ExtendedMatrixDimensions { rows: 2, cols: 128 };
-		let expected = (1..=89).flat_map(|col| (0..=1).map(move |row| (col, row)));
-
-		let result = app_specific_column_cells(&index, &dimensions, 1).unwrap();
-
-		assert_eq!(expected.clone().count(), result.len());
-		result.iter().zip(expected).for_each(|(a, (col, row))| {
-			assert_eq!(a.col, col);
-			assert_eq!(a.row, row);
-		});
-	}
+	// #[test]
```

Comment:
Why are these commented out? Should they be deleted or left for future reference?

Reply:
Meant to be for future reference, but Ill delete them since this is evolving fast. In any case it will be in the history.